helm/APP-NAME/templates/prometheus-get-metrics-rbac.yaml

```diff
@@ -0,0 +1,27 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: prometheus-get-metrics
```

Comment:
Should probably have the [common labels](https://github.com/giantswarm/cluster-resources-app/blob/main/helm/APP-NAME/templates/_helpers.tpl#L18-L31) on both of these resources

Reply:
And to align with how we do names in other templates, you might use:
```handlebars
{{- define "resource.prometheus-get-metrics.name" -}}
prometheus-get-metrics
{{- end -}}
```
and
```yaml
  name: {{ include "resource.prometheus-get-metrics.name" . }}
```
It makes it easier to change the name in the future (might want to add a suffix to avoid a name collision for example) and ensures that the name stays in sync everywhere it's used.

Reply:
Also there's a [default ClusterRole](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#other-component-roles) called `system:monitoring` you might be able to use instead of this.

Reply:
Thanks @tfussell I didn't know about `system:monitoring` but checking that it's more privileged and I'd like to limit what we allow `system:anonymous` to absolute minimum.